app/Jobs/RockTheVote/ImportRockTheVoteRecord.php

```diff
@@ -258,26 +260,101 @@ public function updateUserIfChanged(NorthstarUser $user)
      */
     public function getUserSmsSubscriptionUpdatePayload(NorthstarUser $user)
     {
-        $result = [];
-
         // If registration does not have a mobile provided, there is nothing to update.
         if (! $this->userData['mobile']) {
-            return $result;
+            return [];
         }
 
         // We don't need to update user's SMS subscription if we already did for this registration.
         if (RockTheVoteLog::hasAlreadyUpdatedSmsSubscription($this->record, $user)) {
-            return $result;
+            return [];
+        }
+
+        $payload = $this->parseMobileChangeForUser($user);
+        $payload = array_merge($payload, $this->parseSmsSubscriptionTopicsChangeForUser($user));
+        $payload = array_merge($payload, $this->parseSmsStatusChangeForUser($user));
+
+        return $payload;
+    }
+
+    /**
+     * Returns payload to update mobile if user does not currently have one saved.
+     *
+     * @return array
+     */
+    public function parseMobileChangeForUser(NorthstarUser $user)
+    {
+        $fieldName = 'mobile';
+
+        if ($user->{$fieldName}) {
+            return [];
         }
 
-        // @TODO: Check for changes to user's SMS status and subscription topics.
+        return [$fieldName => $this->userData[$fieldName]];
+    }
+
+    /**
+     * Returns payload to update SMS subscription topics if they have changed.
+     *
+     * @return array
+     */
+    public function parseSmsSubscriptionTopicsChangeForUser(NorthstarUser $user)
+    {
+        $fieldName = 'sms_subscription_topics';
+        $currentSmsTopics = ! empty($user->{$fieldName}) ? $user->{$fieldName} : [];
+
+        // If user opted in to SMS, add the import topics to current topics.
+        if ($this->smsOptIn) {
+            return [$fieldName => array_unique(array_merge($currentSmsTopics, $this->userData[$fieldName]))];
```

Comment:
Should this merge `$currentSmsTopics` with `config('import.rock_the_vote.user.sms_subscription_topics')` instead? What is in `$this->userData[$fieldName]` at this point?

Reply:
Also – this doesn't need to block merging, but it'd be nice to handle this in Northstar with a `SmsTopicController` (like [we do for email subscription topics](https://github.com/DoSomething/northstar/blob/master/app/Http/Controllers/SubscriptionUpdateController.php)) – then we don't need to worry about merging anything in this application (or race-conditions if another change to topics comes in at the same time).

So this could be something like:

```php
public function updateSmsSubscriptionTopicsForUser(NorthstarUser $user)
{
    if ($this->smsOptIn) {
        $this->addSmsTopics($user, config('import.rock_the_vote.user.sms_subscription_topics'));
    } else {
        $this->removeSmsTopics($user, config('import.rock_the_vote.user.sms_subscription_topics'));
    }
}

```

Reply:
> Should this merge $currentSmsTopics with config('import.rock_the_vote.user.sms_subscription_topics') instead? What is in $this->userData[$fieldName] at this point

`$this->userData['sms_subscription_topics]` contains the same `config('import.rock_the_vote.user.sms_subscription_topics')`. It's convenient to store it in the userData for creating a user, where we can just pass the whole array to `northstar->createUser($this->userData)`. For a user that doesn't opt-in, the `$this->userData[sms_subscription_topics]` is an empty array -- which is why it's reading from the config when removing topics.

I did think about the email subscriptions endpoint, but it seemed a bit overkill since we're updating a few different fields for a user per record, and it's cleaner to send a single update request (vs different API requests to modify each field, e.g. subscription, mobile, status, voter registration)